### DISCUSSION TITLE
Vercel REST setted env variables are not visible in Vercel UI

### DISCUSSION THREAD

**Discussion by Dupflo** [vercel/vercel#8702] [Help] [answered]
I am trying to deploy and project and set the env variable at the same time by using v13/deployments. 

```
{
    "name": "test",
    "gitSource": {
        "type": "github",
        "repoId": "XXXXXXX",
        "ref": "main"
    },
    "projectSettings": {
            "installCommand": "npm install",
            "buildCommand": "npm run build",
            "framework": "nextjs"
    },
    "build": {
        "env": {
            "TOKEN": "TESTTOKEN123"
        }
    }
}
```

However I can see the token inside the api response but I am unable to see it inside the Vercel website

<img width="1320" alt="image" src="https://user-images.githubusercontent.com/39828779/194778955-1ebb36d1-fcd1-440c-a5d9-8a0d1ea17eaf.png">

Any help would be great ! Thanks

**Comment by amyegan** [comment, marked as answer]
Sorry for the late response @Dupflo! When you specify env on [/v13/deployments](https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment), it sets the environment variable for only that deployment. It will not be added to Project Settings. If you want to permanently add environment variables to a Project via the API, you would need to use [/v10/projects/{idOrName}/env](https://vercel.com/docs/rest-api#endpoints/projects/create-one-or-more-environment-variables).
